pytest_selenium/drivers/saucelabs.py

```diff
@@ -143,3 +143,38 @@ def _video_html(session):
         id='player{session}'.format(session=session),
         style='border:1px solid #e6e6e6; float:right; height:240px;'
               'margin-left:5px; overflow:hidden; width:320px'))
+
+
+def get_job_url(config, provider, session_id):
+    from datetime import datetime
+
+    job_url = provider.JOB
+    job_auth = config.getini('saucelabs_job_auth').lower()
+
+    if job_auth == 'none':
+        return job_url.format(session=session_id)
+
+    if job_auth == 'token':
+        key = '{0.username}:{0.key}'.format(provider)
+    else:
+        time_format = '%Y-%m-%d'
+        if job_auth == 'hour':
+            time_format = '%Y-%m-%d-%H'
+        elif job_auth != 'day':
+            raise ValueError("Invalid authorization type: {}".format(job_auth))
+
+        ttl = datetime.utcnow().strftime(time_format)
+        key = '{0.username}:{0.key}:{1}'.format(provider, ttl)
+
+    return get_auth_url(job_url, key, session_id)
+
+
+def get_auth_url(url, key, session_id):
+    import hmac
+    from hashlib import md5
+
+    auth_url = url.format(session='{}?auth={}')
```

Comment:
Rather than forcing the parameter into the `session`, you could create a new string based on the resolved job URL (with the session) and the token.